app/questionnaire/questionnaire_store_updater.py

```diff
@@ -38,13 +38,34 @@ def remove_answers(self, answer_ids: List):
         for answer_id in answer_ids:
             self._answer_store.remove_answer(answer_id)
 
+    def add_primary_person_and_answers(self, form, list_name, list_item_id):
+        """
+        If there is already an existing primary person, modify that item.
+        """
+        context = {'primary_person': list_item_id}
+
+        self._list_store.add_list_item(
+            list_name, list_item_id=list_item_id, context=context
+        )
```

Comment:
I think it would be much clearer to have an `add_context` method.

Reply:
Probably not relevant anymore with `primary_person` property/.